### PR TITLE
Be clearer in definition of Inf and avoid relying on Modelica.Constants.

### PR DESCRIPTION
**Pull request by HansOlsson**
This is what I meant with inlining the definition of Modelica.Constants.inf.

For comparison:
In C:
- DBL_MAX is the "maximum finite value of double"; https://en.cppreference.com/w/c/types/limits

In C++  (C++11 and later); https://en.cppreference.com/w/cpp/types/numeric_limits
- std::numeric_limits<double>::max() "returns the largest finite value of double"
- std::numeric_limits<double>::lowest() "returns the lowest finite value of double"

### CODE REVIEW

File: chapters/classes.tex
Comment (beutlich) [inline chapters/classes.tex:1503]: What are _default_ attributes? It can't be default values since there are no values assigned.

Comment (HansOlsson) [reply chapters/classes.tex]: I have tried to rewrite that, but still without introducing a special `+Inf` symbol.
If that fails we can just skip the last two commits.

**Comment by HansOlsson** [comment]
The non-normative comment:
> The minimum representable floating point number is usually minus the maximum representable floating point number, and should not be confused with the minimum positive one.

is related to C++ having such a mismatched pair:

> * std::numeric_limits::max() "returns the largest finite value of double"
> * std::numeric_limits::lowest() "returns the lowest finite value of double"

but I think it is a general naming issue deserving of a comment, and not only a C/C++-problem.

File: chapters/classes.tex
Comment (henrikt-ma) [inline chapters/classes.tex:1481]: Formatting:
```suggestion
  parameter RealType min = $\ldots$, max = $\ldots$; //  See below
```

File: chapters/classes.tex
Comment (henrikt-ma) [inline chapters/classes.tex:1504]: It would be worth pointing out that this still means that infinite values are outside the allowed range:
```suggestion
Therefore they only constrain the value to be finite, i.e., floating point representations of infinity are not allowed.
```

File: chapters/classes.tex
Comment (henrikt-ma) [inline chapters/classes.tex:1506]: I'm not sure _minus_ should be used like that; this looks better to me:
```suggestion
The minimum representable floating point number is usually the negation of the maximum representable floating point number, and should not be confused with the minimum positive one.
```

File: chapters/classes.tex
Comment (HansOlsson) [reply chapters/classes.tex:1504]: I will reformulate - in my opinion the constraint to be an actual finite number is part of the type, and the default min/max do not restrict that further.

File: chapters/classes.tex
Comment (henrikt-ma) [reply chapters/classes.tex]: But why have anything at all, considering that no default is given for `start`?
```suggestion
  parameter RealType min; // Minimum allowed value.
  parameter RealType max; // Maximum allowed value.
```

Comment (henrikt-ma) [reply chapters/classes.tex]: By the way, why do some of the attributes have explanations in comments, while some (`unit` and `displayUnit`) have description strings?

Comment (HansOlsson) [reply chapters/classes.tex]: > But why have anything at all, considering that no default is given for `start`?

I believe I tried that earlier, and @beutlich complained about that variant.
Since we have the assert using `min` I can see that it might require more.

As for `unit `and `displayUnit `having description strings and some attributes having comments: it has been that way for some time, I agree that it should likely be cleaned up at some point - it's just that we need reviews for existing topics before opening additional ones.

(Oh, and `start` not having any value was a very deliberate choice.)

File: chapters/classes.tex
Comment (henrikt-ma) [inline chapters/classes.tex:1481]: Perhaps the _See below_ goes without saying, and it would be of more interest to explain that these limit the value?
```suggestion
  parameter RealType min = $\ldots$; // Lower bound for $\langle$$\mbox{\emph{value}}$$\rangle$
  parameter RealType max = $\ldots$; // Upper bound for $\langle$$\mbox{\emph{value}}$$\rangle$
```

Also, if we now introduce the use of ellipsis for a default being explained in the text, shouldn't `fixed` be handled in the same style?
```
  parameter BooleanType fixed = $\ldots$; // Enforce exact value of 'start'
```
together with
> The default \lstinline!fixed!-attribute is \lstinline!true! for parameters and constants, and \lstinline!false! for other variables.

File: chapters/classes.tex
Comment (henrikt-ma) [inline chapters/classes.tex:1505]: Just for better source code readability:
```suggestion

\begin{nonnormative}
```

**Review by henrikt-ma — CHANGES_REQUESTED** [review]
If we start using ellipsis for explaining default attributes which cannot be expressed nicely in the pseudo code, we should use this style for `fixed` too.

File: chapters/classes.tex
Comment (HansOlsson) [reply chapters/classes.tex:1481]: Sure

File: chapters/classes.tex
Comment (HansOlsson) [reply chapters/classes.tex:1505]: Did that as part of update

File: chapters/classes.tex
Comment (henrikt-ma) [inline chapters/classes.tex:1503]: I think we tend do use suspended hyphenation.  Also, why not use `RealType`?
```suggestion
The default \lstinline!min!- and \lstinline!max!-attributes are the minimum and maximum representable finite floating point numbers of \lstinline!RealType!.
```

File: chapters/classes.tex
Comment (henrikt-ma) [inline chapters/classes.tex:1504]: I guess you mean this:
```suggestion
The $\langle$$\mbox{\emph{value}}$$\rangle$ must be a finite floating point number, and thus the default \lstinline!min!- and \lstinline!max!-attributes do not impose any further constraints.
```

File: chapters/classes.tex
Comment (henrikt-ma) [inline chapters/classes.tex:1507]: Perhaps better to avoid _one_ in a complicated sentence like this?
```suggestion
Of the representable floating point numbers, the minimum number is usually the negation of the maximum number, and should not be confused with the minimum positive number.
```

File: chapters/classes.tex
Comment (henrikt-ma) [inline chapters/classes.tex:1556]: As above:
```suggestion
The default \lstinline!min!- and \lstinline!max!-attributes are the minimum and maximum numbers of \lstinline!IntegerType!.
```

**Review by henrikt-ma — CHANGES_REQUESTED** [review]
Only requesting minor changes of wording and formatting fixes.

**Review by henrikt-ma — APPROVED** [review]
Looks good to me.  (Would be nice to have one more approval before it's merged.)

**Comment by henrikt-ma** [comment]
I guess we should open a PR for aligning the definition of `Modelica.Constants.inf` as soon as this has been merged.

It might also be a good idea to keep the future description of `Modelica.Constants.inf` in mind when considering the changes brought by this PR.

Direct use of the new formulations in the specification would make it look like this, which is problematic because users typically are not aware of `RealType`:
> "Maximum representable finite floating point number of RealType"

However, to me it would not be a big problem if the description of `Modelica.Constants.inf` diverged a little bit by referring to `Real` instead of `RealType`:
> "Maximum allowed Real value"

**Pull request merged**
